component/otelcol/processor/k8sattributes/k8sattributes.go

```diff
@@ -0,0 +1,257 @@
+// Package attributes provides an otelcol.processor.attributes component.
+package k8sattributes
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/processor"
+	"github.com/mitchellh/mapstructure"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "otelcol.processor.k8sattributes",
+		Args:    Arguments{},
+		Exports: otelcol.ConsumerExports{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := k8sattributesprocessor.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+type FieldExtractConfig struct {
+	TagName  string `river:"tag_name,attr,optional"`
+	Key      string `river:"key,attr,optional"`
+	KeyRegex string `river:"key_regex,attr,optional"`
+	Regex    string `river:"regex,attr,optional"`
+	From     string `river:"from,attr,optional"`
+}
+
+func (args FieldExtractConfig) convert() map[string]interface{} {
+	return map[string]interface{}{
+		"tag_name":  args.TagName,
+		"key":       args.Key,
+		"key_regex": args.KeyRegex,
+		"regex":     args.Regex,
+		"from":      args.From,
+	}
+}
+
+type ExtractConfig struct {
```

Comment:
It'd be good to move all the config structs and their methods to a new file called types.go.